fix: check for spatialite-tools at configure time (#6159)
valhalla/valhalla: test/CMakeLists.txt

## Test-specific data, properties and dependencies
target_compile_definitions(logging PRIVATE LOGGING_LEVEL_ALL)

# tz.sqlite is built by scripts/valhalla_build_timezones, which shells out to the

> # spatialite-tools CLI programs. Fail clearly at configure time if they are missing.

Reviewer: too much text IMO
```suggestion
# make sure we can build tz.sqlite
```

> if(ENABLE_DATA_TOOLS)

Author: Put this in the test CMake and gated it on ENABLE_DATA_TOOLS so it's skipped when tests are off, which is what the issue asked. The existing check only covers the libspatialite library, not the spatialite_tool/spatialite programs the timezone script runs. That was the gap.

Reviewer: the right place to put this is the root cmakelists.txt, bcs it's surely not only a problem for _tests_ but for everything.

  find_program(SPATIALITE_TOOL_EXECUTABLE NAMES spatialite_tool)
  find_program(SPATIALITE_EXECUTABLE NAMES spatialite)
  if(NOT SPATIALITE_TOOL_EXECUTABLE OR NOT SPATIALITE_EXECUTABLE)

>     message(FATAL_ERROR "spatialite-tools not found: need both 'spatialite_tool' and 'spatialite' to build tz.sqlite for tests. Install with: brew install spatialite-tools (macOS) or apt-get install spatialite-bin (Debian/Ubuntu).")

Reviewer: nit: also too wordy IMO (and its verbosity not accurate, there's a zillion pkg managers out there). maybe jsut 
```suggestion
    message(FATAL_ERROR "spatialite-tools not found: need both 'spatialite_tool' and 'spatialite', install the correct package for your system.")
```
